python/ray/data/dataset.py

```diff
@@ -114,6 +116,7 @@ def __init__(self, blocks: BlockList, epoch: int, stats: DatasetStats):
         self._uuid = uuid4().hex
         self._epoch = epoch
         self._stats = stats
+        self._name = name
```

Comment:
Agreed on falling back to a truncated (~8 characters?) prefix of the uuid for the name here! Something like:
```python
self._name = name or f"dataset-{self._uuid[:8]}"
```